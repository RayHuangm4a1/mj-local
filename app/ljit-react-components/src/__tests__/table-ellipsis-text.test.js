import React from 'react';
import { mount, shallow, } from 'enzyme';
import TableEllipsisText from '../components/table-ellipsis-text';

describe('table-ellipsis-text', () => {

	it('handle default props', () => {
		const {
			tooltipWidth,
			tooltipPosition,
		} = TableEllipsisText.defaultProps;

		expect(tooltipWidth).toEqual(350);
		expect(tooltipPosition).toEqual(TableEllipsisText.TooltipPositionEnums.BOTTOM);
	});
	it('should renders correctly', () => {
		const wrapper = shallow(
			<TableEllipsisText
				text="mock-text"
				className="mock-classname"
				tooltipWidth={200}
				positionToRight={10}
				positionToTop={10}
				tooltipPosition={TableEllipsisText.TooltipPositionEnums.RIGHT}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const text = 'mock-text';
		const className = 'mock-classname';
		const positionToRight = 10;
		const positionToTop = 20;
		const tooltipWidth = 300;
		const tooltipPosition = TableEllipsisText.TooltipPositionEnums.RIGHT;
		const wrapper = mount(
			<TableEllipsisText
				text={text}
				className={className}
				tooltipWidth={tooltipWidth}
				positionToRight={positionToRight}
				positionToTop={positionToTop}
				tooltipPosition={tooltipPosition}
			/>
		);

		expect(wrapper.props().text).toEqual(text);
		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().positionToRight).toEqual(positionToRight);
		expect(wrapper.props().positionToTop).toEqual(positionToTop);
		expect(wrapper.props().tooltipWidth).toEqual(tooltipWidth);
		expect(wrapper.props().tooltipPosition).toEqual(tooltipPosition);

	});

	it('should be selectable by class ljit-table-ellipsis-text', () => {
		const wrapper = shallow(<TableEllipsisText />);

		expect(wrapper.hasClass('ljit-table-ellipsis-text')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<TableEllipsisText className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	describe('Tooltip Position Enums', () => {
		it('should contain left property.', () => {
			expect(TableEllipsisText.TooltipPositionEnums).toHaveProperty('LEFT', 'left');
		});


		it('should contain right property.', () => {
			expect(TableEllipsisText.TooltipPositionEnums).toHaveProperty('RIGHT', 'right');
		});

		it('should contain top property.', () => {
			expect(TableEllipsisText.TooltipPositionEnums).toHaveProperty('TOP', 'top');
		});

		it('should contain bottom property.', () => {
			expect(TableEllipsisText.TooltipPositionEnums).toHaveProperty('BOTTOM', 'bottom');
		});
	});
});

