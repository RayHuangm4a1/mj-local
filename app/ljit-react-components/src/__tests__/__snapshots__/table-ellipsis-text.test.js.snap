// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table-ellipsis-text should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableEllipsisText
    className="mock-classname"
    positionToRight={10}
    positionToTop={10}
    text="mock-text"
    tooltipPosition="right"
    tooltipWidth={200}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-table-ellipsis-text__text"
        >
          mock-text
        </div>,
        <span
          className="ljit-table-ellipsis-text__tooltips ljit-table-ellipsis-text__tooltips--right"
          style={
            Object {
              "right": "10px",
              "top": "10px",
              "width": "200px",
            }
          }
        >
          mock-text
        </span>,
      ],
      "className": "ljit-table-ellipsis-text mock-classname",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mock-text",
          "className": "ljit-table-ellipsis-text__text",
        },
        "ref": null,
        "rendered": "mock-text",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mock-text",
          "className": "ljit-table-ellipsis-text__tooltips ljit-table-ellipsis-text__tooltips--right",
          "style": Object {
            "right": "10px",
            "top": "10px",
            "width": "200px",
          },
        },
        "ref": null,
        "rendered": "mock-text",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-table-ellipsis-text__text"
          >
            mock-text
          </div>,
          <span
            className="ljit-table-ellipsis-text__tooltips ljit-table-ellipsis-text__tooltips--right"
            style={
              Object {
                "right": "10px",
                "top": "10px",
                "width": "200px",
              }
            }
          >
            mock-text
          </span>,
        ],
        "className": "ljit-table-ellipsis-text mock-classname",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mock-text",
            "className": "ljit-table-ellipsis-text__text",
          },
          "ref": null,
          "rendered": "mock-text",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mock-text",
            "className": "ljit-table-ellipsis-text__tooltips ljit-table-ellipsis-text__tooltips--right",
            "style": Object {
              "right": "10px",
              "top": "10px",
              "width": "200px",
            },
          },
          "ref": null,
          "rendered": "mock-text",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
